### PR TITLE
CLN: address TODOs, FIXMEs

### PR DESCRIPTION
**Pull request by jbrockmendel**
Will have to see what the CI says about the cython flake8 config change. <b>update</b> cython flake8 config moved to #44260

### CODE REVIEW

File: pandas/_libs/tslibs/fields.pyx [hunk 1]
```diff
@@ -198,7 +198,7 @@ cdef inline bint _is_on_month(int month, int compare_month, int modby) nogil:
 @cython.wraparound(False)
 @cython.boundscheck(False)
 def get_start_end_field(const int64_t[:] dtindex, str field,
-                        object freqstr=None, int month_kw=12):
+                        str freqstr=None, int month_kw=12):
```
Comment (twoertwein) [inline pandas/_libs/tslibs/fields.pyx:201]: I'm not familiar with typing in cython, but I would assume that the default value has to be a string for `freqstr`.

Comment (jbrockmendel) [reply pandas/_libs/tslibs/fields.pyx:201]: None is allowed in this context

File: pandas/_libs/join.pyx [hunk 2]
```diff
@@ -478,7 +494,7 @@ def inner_join_indexer(ndarray[numeric_object_t] left, ndarray[numeric_object_t]
             if lval == rval:
                 lindexer[count] = i
                 rindexer[count] = j
-                result[count] = rval
+                result[count] = lval
```
Comment (jreback) [inline pandas/_libs/join.pyx:497]: why this change, this didn't break anything?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/join.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/join.pyx:497]: lval == rval here, this just makes this block match whats done elsewhere

**Review by jreback — APPROVED** [review]
